portal/assignment.py

```diff
@@ -292,3 +281,72 @@ def assignment_grades():
 
         return render_template('layouts/teacher/assignments/assignment-grades.html', assignment=assignment, assignment_name=assignment_name)
     return redirect(url_for('teacher.home'))
+@bp.route('grades/grade-book', methods=('GET', 'POST'))
+@login_required
+@admin
+def grade_view():
+    if request.method == 'POST':
+        session = request.form['gradebook']
+        with db.get_db() as con:
+            with con.cursor() as cur:
+                cur.execute("""
+                SELECT s.id, u.id AS user_id, u.first_name, u.last_name
+                FROM sessions s JOIN roster r
+                ON s.id = r.session_id
+                JOIN users u
+                ON r.student_id = u.id
+                WHERE s.id = %s
+                """, (session))
+                students = cur.fetchall()
+        return render_template('grade/gradebook.html', students=students)
+    return redirect(url_for('teacher.sessions'))
+@bp.route('grades/all-grades', methods=('GET', 'POST'))
+@login_required
+@admin
+def personal_grades():
+    if request.method == 'POST':
+        student = request.form['student_id']
+        session = request.form['session_id']
+        with db.get_db() as con:
+            with con.cursor() as cur:
+                cur.execute("""
+                SELECT u.first_name, u.last_name, g.grades, a.name, a.description, a.points
+                FROM session_assignments s JOIN roster r
+                ON s.session_id = r.session_id
+                JOIN users u
+                ON r.student_id = u.id
+                JOIN assignments a
+                ON s.assignment_id = a.id
+                JOIN assignment_grades g
+                ON g.assigned_id = s.work_id
+                WHERE s.session_id = %s AND u.id = %s
+                """, (session, student))
+                grades = cur.fetchall()
```

Comment:
it seems like you fetch all the necessary information here and then write smaller queries to select parts of the same information again below, is that necessary? it seems like the `grades` variable should already have all of the information you need, or that you could adjust the query to get it instead of running several back-to-back